### ISSUE TITLE
I cannot connect and pull

### ISSUE THREAD

**Issue by pjindras** [Konstanius/MineCICD#7]
![image](https://github.com/user-attachments/assets/8b24839e-44e7-4834-982e-0b1d45f3e0ed)

```
11:01:04[ERROR] [MineCICD] Failed to pull changes
11:01:04[ERROR] [MineCICD] Cannot run program "which": error=2, No such file or directory
java.io.IOException: Cannot run program "which": error=2, No such file or directory
at java.base/java.lang.ProcessBuilder.start(Unknown Source) ~[?:?]
at java.base/java.lang.ProcessBuilder.start(Unknown Source) ~[?:?]
at java.base/java.lang.Runtime.exec(Unknown Source) ~[?:?]
at java.base/java.lang.Runtime.exec(Unknown Source) ~[?:?]
at java.base/java.lang.Runtime.exec(Unknown Source) ~[?:?]
at MineCICD-2.1.1.jar/ml.konstanius.minecicd.GitSecret.configureGitSecretFiltering(GitSecret.java:133) ~[MineCICD-2.1.1.jar:?]
at MineCICD-2.1.1.jar/ml.konstanius.minecicd.GitUtils.pull(GitUtils.java:277) ~[MineCICD-2.1.1.jar:?]
at MineCICD-2.1.1.jar/ml.konstanius.minecicd.BaseCommand.lambda$onCommand$0(BaseCommand.java:211) ~[MineCICD-2.1.1.jar:?]
at org.bukkit.craftbukkit.scheduler.CraftTask.run(CraftTask.java:91) ~[leaf-1.21.jar:1.21-DEV-3619782]
at org.bukkit.craftbukkit.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57) ~[leaf-1.21.jar:1.21-DEV-3619782]
at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[leaf-1.21.jar:?]
at java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(Unknown Source) ~[?:?]
at java.base/java.lang.VirtualThread.run(Unknown Source) ~[?:?]
Caused by: java.io.IOException: error=2, No such file or directory
at java.base/java.lang.ProcessImpl.forkAndExec(Native Method) ~[?:?]
at java.base/java.lang.ProcessImpl.<init>(Unknown Source) ~[?:?]
at java.base/java.lang.ProcessImpl.start(Unknown Source) ~[?:?]
... 13 more
11:01:04[ERROR] [MineCICD] java.base/java.lang.ProcessBuilder.start(Unknown Source)
java.base/java.lang.ProcessBuilder.start(Unknown Source)
java.base/java.lang.Runtime.exec(Unknown Source)
java.base/java.lang.Runtime.exec(Unknown Source)
java.base/java.lang.Runtime.exec(Unknown Source)
MineCICD-2.1.1.jar//ml.konstanius.minecicd.GitSecret.configureGitSecretFiltering(GitSecret.java:133)
MineCICD-2.1.1.jar//ml.konstanius.minecicd.GitUtils.pull(GitUtils.java:277)
MineCICD-2.1.1.jar//ml.konstanius.minecicd.BaseCommand.lambda$onCommand$0(BaseCommand.java:211)
org.bukkit.craftbukkit.scheduler.CraftTask.run(CraftTask.java:91)
org.bukkit.craftbukkit.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57)
com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22)
java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(Unknown Source)
java.base/java.lang.VirtualThread.run(Unknown Source)
```

Maybe I missed some steps, but I did all as it is in provided tutorial.

**Comment by Konstanius** [comment]
References https://github.com/Konstanius/MineCICD/blob/68404d766bbed6ca397ced764604731da1eeb5e5/src/main/java/ml/konstanius/minecicd/GitSecret.java#L133

You are probably using Pterodactyl Panel or something similar which uses Docker with barebone Linux under the hood, correct?

**Comment by Konstanius** [comment]
Fixed in latest release Version 2.1.2
https://github.com/Konstanius/MineCICD/releases/tag/2.1.2-Release

Please download the new version and try again

**Comment by Konstanius** [comment]
Since this issue has been inactive and a fix was pushed, I am marking it as done.
